scripts/weather.py

```diff
@@ -0,0 +1,62 @@
+import requests, json
+from datetime import datetime, timedelta
+from deepmerge import always_merger
+from scripts.config import *
+from google.cloud import storage
+
+
+def get_weather_data(query, limit=30):
+    weather = {}
+    weather["forecast"] = {}
+    for day in range(limit, -1, -1):
+        date = datetime.today() - timedelta(days=day)
+        full_url = BASE_URL.format(API_KEY, query, date.strftime('%Y-%m-%d'))
+        response = requests.get(full_url)
+        data = response.json()
+        try:
+            forecast = data["forecast"]["forecastday"][0]
+            location = data["location"]
+            forecast["day"].pop("condition")
+            weather = {**weather, **location}
+
+            weather["forecast"][forecast["date_epoch"]] = forecast["day"]
+        except :
+            continue
+
+    return weather
+
+def __load_state_file(state):
+    try:
+        with open("{}{}.json".format(DATA_PATH, state)) as f:
+            data = json.load(f);
+            return data
+    except FileNotFoundError:
+        return {}
+
+def __update_state_file(state, data):
+    with open("{}{}.json".format(DATA_PATH, state), "w+") as f:
+        json.dump(data, f)
+
+def get_state_weather_locally(state):
```

Comment:
This was good for debug mode, but once the ETL works, we can probably remove the local access in favor of always reading from the bucket, similar to the visit data.